src/components/input.tsx

```diff
@@ -0,0 +1,64 @@
+import { type VariantProps, cva } from 'class-variance-authority'
+import React, { forwardRef } from 'react'
+
+import { cn } from '@/utils/cn'
+
+const inputVariants = cva(
+  [
+    'w-full h-14 rounded-xl border border-2 px-4 text-base text-medium',
+    'focus:outline-none focus:border-focus',
+    'placeholder-disabled',
+  ],
+
+  {
+    variants: {
+      variant: {
+        default: 'border-base',
+      },
+      error: {
+        true: 'border-error',
+      },
+      loading: {
+        true: 'text-transparent',
+      },
+      hasIcon: {
+        true: 'pl-12',
+      },
+      disabled: {
+        true: 'bg-disabled border-transparent',
+      },
+    },
+    defaultVariants: {
+      variant: 'default',
+    },
+  },
+)
+
+export interface InputProps
+  extends VariantProps<typeof inputVariants>,
+    React.InputHTMLAttributes<HTMLInputElement> {
+  loading?: boolean
+  error?: boolean
+  disabled?: boolean
+  icon?: React.ReactNode
+  type?: 'text' | 'password' | 'email' | 'number'
+}
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(function Input(
```

Comment:
If we pass an error message we need to:
- render the error message below the input
- add the `aria-invalid` and `aria-describedby` attributes to the input

Additionally we can use the label component directly, by adding a `label` prop:
```tsx
<Input label="Amount" {...restOfProps} />
```

Reply:
we must use it separately because there is a situation when we have also a link besides the actualy label (as per graphic)
e.g. Label [link](url)